api/types/resource_153.go

```diff
@@ -23,6 +23,12 @@ import (
 	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
 )
 
+// ResourceMetadata is the smallest interface that defines a Teleport resource.
+type ResourceMetadata interface {
```

Comment:
@rosstimothy, I've snuck this in here based on your suggestions, in the hopes that because it exists it might "catch up" and be used (instead of the larger Resource153).

This isn't strictly necessary, so I could drop it. Let me know your thoughts.

Reply:
Part of me feels like this may be too minimalist - it feels like to me that the ability to determine the Kind/Version of an generic resource is potentially more important than determining the name of it - but perhaps my understanding of the cases where we handle resources generically is too limited.

Reply:
I think that being minimalistic is the idea. I would rather see smaller in-promptu interfaces than keep adding to this one, or preemptively defining more.

Another alternative would be to move all we care about into Metadata (which I'd be tempted to do if we were working on a new design).

I'll go ahead and move forward as-is, but happy to retract this if y'all prefer otherwise.